apps/email/js/cards/lst/edit_controller.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+define(function(require, exports) {
+  var cards = require('cards'),
+      ConfirmDialog = require('confirm_dialog'),
+      deleteConfirmMsgNode = require('tmpl!../msg/delete_confirm.html'),
+      mozL10n = require('l10n!'),
+      toaster = require('toaster');
+
+  return {
+    createdCallback: function() {
+      this.editMode = false;
+      this.selectedMessages = null;
+      this.editModeEnabled = false;
+    },
+
+    setEditMode: function(editMode) {
+      // Do not bother if edit mode is not enabled yet.
+      if (!this.editModeEnabled) {
+        return;
+      }
+
+      this._setEditMode(editMode);
+    },
+
+    // This function is called from setEditMode() after ensuring that
+    // the backend is in a state where we can safely use edit mode.
+    _setEditMode: function(editMode) {
+      var i;
+
+      this.editMode = editMode;
+
+      // XXX the manual DOM play here is now a bit overkill; we should very
+      // probably switch top having the CSS do this for us or at least invest
+      // some time in cleanup.
+      if (editMode) {
+        this.classList.add('show-edit');
+
+        this.selectedMessages = [];
+        this.selectedMessagesUpdated();
+      }
+      else {
+        this.classList.remove('show-edit');
+
+        this.selectedMessages = null;
+      }
+
+      // Reset checked mode for all message items.
+      var msgNodes = this.msgVScroll.querySelectorAll('.msg-header-item');
+      for (i = 0; i < msgNodes.length; i++) {
+        this.setMessageChecked(msgNodes[i], false);
+      }
+
+      if (this.editModeChanged) {
+        this.editModeChanged(editMode);
+      }
+    },
+
+    // Event handler wired up in HTML template
+    setEditModeStart: function() {
+      this.setEditMode(true);
+    },
+
+    // Event handler wired up in HTML template
+    editHeaderClose: function() {
+      this.setEditMode(false);
+    },
+
+    /**
+     * Set or unset the select state based on the edit mode.
+     */
+    setSelectState: function(msgNode, message) {
```

Comment:
(I know this just just moved code, but) maybe we can rename this to updateDomSelectedState or reflectDomSelectedState or something like that since this is really about reflecting existing internal state.  Whereas "set" makes it seem like we're manipulating internally tracked state.

Ditto for setMessageChecked below, which it also happens to call.  Especially with the manipulation of the checked state, it makes it seem like we care about the DOM state, but we don't.  The true state is in `selectedMessages`.